test/runtests.jl

```diff
@@ -61,3 +61,196 @@ cd(Pkg.dir("Coverage")) do
     @test isempty(Coverage.process_cov("fakefile",datadir))
     rm("fakefile")
 end
+
+
+
+
+######################
+# codecovio.jl tests #
+######################
+
+"""
+extracts the api URL from stdout in a codecov.io submit call
+very helpful for testing codecovio.jl
+"""
+function extract_codecov_url(fun)
+    originalSTDOUT = STDOUT
+
+    (outRead, outWrite) = redirect_stdout()
+
+    fun()
+
+    close(outWrite)
+
+    data = utf8(readavailable(outRead))
+
+    close(outRead)
+    redirect_stdout(originalSTDOUT)
+
+    lines = split(data, "\n")
+
+    url = "None"
+    get_next = false
+    for line in lines
+        if get_next 
+            url = line
+            get_next = false
+        end
+        if contains(line, "Codecov.io API URL")
+            get_next = true
+        end
+    end
+
+    #println("url: $(url)")
+    @assert url != "None" "unable to find codecov api url in stdout, check for changes in codecovio.jl" 
+    return url
+end
+
+
+# clear system defined ENV vars in preparation for testing
+
+delete!(ENV, "CODECOV_URL")
+delete!(ENV, "CODECOV_TOKEN")
```

Comment:
would probably be better to use `withenv` for temporarily clearing or setting these and automatically restoring them